diff --git a/lib/node/transform.js b/lib/node/transform.js
index 11bdf38db57a3900cc3ce127bf886de595db938f..e47e518872eb1413dcba71f8d0291aaadbafaacb 100644
--- a/lib/node/transform.js
+++ b/lib/node/transform.js
@@ -198,15 +198,19 @@ function toSourceOffset(sourceScript, language, serviceScript, position, filter)
     }
 }
 function* toSourceRanges(sourceScript, language, serviceScript, start, end, fallbackToAnyMatch, filter) {
+    if(start > 0 || end > 0){
+      start -= getMappingOffset(language, serviceScript)
+      end -= getMappingOffset(language, serviceScript)
+    }
     if (sourceScript) {
         const map = language.maps.get(serviceScript.code, sourceScript);
-        for (const [sourceStart, sourceEnd] of map.toSourceRange(start - getMappingOffset(language, serviceScript), end - getMappingOffset(language, serviceScript), fallbackToAnyMatch, filter)) {
+        for (const [sourceStart, sourceEnd] of map.toSourceRange(start, end, fallbackToAnyMatch, filter)) {
             yield [sourceScript.id, sourceStart, sourceEnd];
         }
     }
     else {
         for (const [sourceScript, map] of language.maps.forEach(serviceScript.code)) {
-            for (const [sourceStart, sourceEnd] of map.toSourceRange(start - getMappingOffset(language, serviceScript), end - getMappingOffset(language, serviceScript), fallbackToAnyMatch, filter)) {
+            for (const [sourceStart, sourceEnd] of map.toSourceRange(start, end, fallbackToAnyMatch, filter)) {
                 yield [sourceScript.id, sourceStart, sourceEnd];
             }
         }
