diff --git a/lib/quickstart/runTsc.js b/lib/quickstart/runTsc.js
index 5d1a9ce95c19893efd3903a0cb551bb72321e28d..4b88f491602f85c29fa9760b7dea5db8ad9fe5f5 100644
--- a/lib/quickstart/runTsc.js
+++ b/lib/quickstart/runTsc.js
@@ -71,7 +71,7 @@ function transformTscContent(tsc, proxyApiPath, extraSupportedExtensions, extraE
     // proxy createProgram
     tsc = replace(tsc, /function createProgram\(.+\) {/, s => `var createProgram = require(${JSON.stringify(proxyApiPath)}).proxyCreateProgram(`
         + [
-            `new Proxy({}, { get(_target, p, _receiver) { return eval(p); } } )`,
+            `require('typescript')`,
             `_createProgram`,
             `require(${JSON.stringify(getLanguagePluginsFile)}).getLanguagePlugins`,
         ].join(', ')
